source/shared/cpp/ObjectModel/MarkDownParser.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include <iomanip>
+#include <list>
+#include "pch.h"
+#include "BaseCardElement.h"
+#include "Enums.h"
+
+namespace AdaptiveCards
+{
+class MarkDownParser
+{
+public:
+    MarkDownParser(const std::string &txt);
+
+    enum DelimiterType
+    {
+        Init,
+        Alphanumeric,
+        Puntuation,
+        Escape,
+        WhiteSpace,
+        Underscore = 0x8,
+        Asterisk   = 0x10,
+    };
+
+    enum EmphasisType
+    {
+        EmphasisItalic = 1,
+        EmphasisBold = 2,
+    };
+
+    enum EmphasisState
+    {
+        EmphasisDetected = 0x1,
+        EmphasisDetecting = 0x2,
+        EmphasisOn = EmphasisDetected | EmphasisDetecting,
+        EmphasisEnded = 0x4,
+        EmphasisNotDetected = 0x8,
+        EmphasisOff = EmphasisEnded | EmphasisNotDetected,
+    };
+
+    struct Emphasis
+    {
+        int m_emphCnts;
+        unsigned m_idx;
+        std::vector<int> m_tags;
+        Emphasis(unsigned int empCnts, unsigned idx) : 
+            m_emphCnts(empCnts), m_idx(idx), m_tags(){};
+    };
+
+    bool IsLeftEmphasisDelimiter(void);
```

Comment:
>void [](start = 33, length = 4)

Avoid using void inside parameter list.

bool function() is the preferred way.